src/lomc.jl

```diff
@@ -432,86 +207,32 @@ function lomc!(
     v = default_starting_vector(address; style, threading, initiator)
     return lomc!(ham, v; address, kwargs...) # pass address for setting the default shift
 end
-# For continuation, you can pass a QMCState and a DataFrame
-function lomc!(state::QMCState, df=DataFrame(); laststep=0, name="lomc!", metadata=nothing)
-    if !iszero(laststep)
-        state.laststep = laststep
-    end
 
-    # initialise report
-    report = Report()
-    report_default_metadata!(report, state)
-    isnothing(metadata) || report_metadata!(report, metadata) # add user metadata
-
-    # Sanity checks.
-    step, laststep = state.step, state.laststep
-    for replica in state.replicas
-        @assert replica.params.step == step
-        @assert replica.params.laststep == laststep
-    end
-    check_transform(state.replica, state.hamiltonian)
-
-    # main loop
-    initial_step = step
-    update_steps = max((laststep - initial_step) ÷ 200, 100) # log often but not too often
-    # update_steps = 400
-    @withprogress name=name while step < laststep
-        step += 1
-        if step % reporting_interval(state.r_strat) == 0
-            report!(state.r_strat, step, report, :steps, step)
-        end
-        # This loop could be threaded if num_threads() == num_replicas? MPIData would
-        # need to be aware of the replica's id and use that in communication.
-        success = true
-        for replica in state.replicas
-            success &= advance!(report, state, replica)
-        end
-        if step % reporting_interval(state.r_strat) == 0
-            replica_names, replica_values = replica_stats(state.replica, state.replicas)
-            report!(state.r_strat, step, report, replica_names, replica_values)
-            report_after_step(state.r_strat, step, report, state)
-            ensure_correct_lengths(report)
-        end
-        if step % update_steps == 0 # for updating progress bars
-            @logprogress (step-initial_step)/(laststep-initial_step)
-        end
-        !success && break
-    end
-
-    # Put report into DataFrame and merge with `df`. We are assuming the previous `df` is
-    # compatible, which should be the case if the run is an actual continuation. Maybe the
-    # DataFrames should be merged in a more permissive manner?
-    result_df = finalize_report!(state.r_strat, report)
-    if !isempty(df)
-        df = vcat(df, result_df) # metadata is not propagated
-        for (key, val) in get_metadata(report) # add metadata
-            DataFrames.metadata!(df, key, val)
-        end
-        return (; df, state)
-    else
-        return (; df=result_df, state)
-    end
+function lomc!(::AbstractMatrix, v=nothing; kwargs...)
+    throw(ArgumentError("Using lomc! with a matrix is no longer supported. Use `MatrixHamiltonian` instead."))
 end
 
+
 """
```

Comment:
I'd move this function (and maybe `FCIQMCTransitionOperator` as well) to the file that defines the states. I think lomc.jl should only include the legacy interface.